TST: insert 'match' to bare pytest raises in pandas/tests/tseries/off…

krajatcl opened the pull request:
…sets/test_offsets.py

- [ ] ref https://github.com/pandas-dev/pandas/issues/30999
- [ ] tests added / passed
- [ ] passes `black pandas`
- [ ] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

jbrockmendel commented on pandas/tests/tseries/offsets/test_offsets.py:
```diff
@@ -4315,7 +4326,8 @@ def test_valid_month_attributes(kwd, month_classes):
     cls = month_classes
     # check that we cannot create e.g. MonthEnd(weeks=3)
-    with pytest.raises(TypeError):
+    msg = f"__init__\\(\\) got an unexpected keyword argument '{kwd}'"
```
can you make these single-slash and use "rf" prefix

jbrockmendel commented:
nitpick, otherwise LGTM

jbrockmendel commented on pandas/tests/tseries/offsets/test_offsets.py:
```diff
@@ -4338,32 +4350,34 @@ def test_valid_tick_attributes(kwd, tick_classes):
     cls = tick_classes
     # check that we cannot create e.g. Hour(weeks=3)
-    with pytest.raises(TypeError):
+    msg = f"__init__\\(\\) got an unexpected keyword argument '{kwd}'"
```
same here

krajatcl commented:
Hi can you add this label 
hacktoberfest-accepted

jreback commented:
hmm can you merge master and ping on green.

krajatcl commented:
I did git reset with upstream/master and than ran pytest pandas/tests/tseries/offsets/test_offsets.py  -s. And it showed lots of Exceptions raised (293 failed) should i ignore them for now or need to fix them too along with match in pytest.raises.
@jreback

jreback commented:
> I did git reset with upstream/master and than ran pytest pandas/tests/tseries/offsets/test_offsets.py -s. And it showed lots of Exceptions raised (293 failed) should i ignore them for now or need to fix them too along with match in pytest.raises.
> @jreback

really? you should have none

krajatcl commented:
Yes, There are and i have not even made any changes in upstream/master at that point. I am not sure whether it has something to do with my local setup or not. But if i remember correctly at start of this pull request i saw none.

@jreback

jreback commented:
> Yes, There are and i have not even made any changes in upstream/master at that point. I am not sure whether it has something to do with my local setup or not. But if i remember correctly at start of this pull request i saw none. I will check with my local setup once again and see if i made any mistake there but i am using same docker setup.
> 
> @jreback

ok pls just merge master and ping on green

jreback commented:
yeah this is actually failing, so please see if you can fixup things.

Commit:
TST: insert 'match' to bare pytest raises in pandas/tests/tseries/offsets/test_offsets.py

krajatcl commented:
hi @jreback 
I am new to open source so earlier i only used to do **pytest {corresponding file name which i changed}** for some of my pull request raised. But as i noticed in this request that some of test cases were failing due to one of my previous PR **https://github.com/pandas-dev/pandas/pull/37027.** On that i did same basic testing which passed but caused failure in testing env. I have fixed those issues in this comment. So can you guide me (or any documentation link) what else i need to do to make sure that my changes don't cause problem later on.
Other command i found in the testing env was 
**pytest -m "not slow and not network and not clipboard" -n auto --dist=loadfile -s --strict --durations=30 --junitxml=test-data.xml pandas**. 
Is this enough for initial testing.
Sorry to bother you with this request but any help will be much appreciated.

jreback commented:
i think master is now updated, so need to merge master and see if there is anything left to change.

krajatcl commented:
> i think master is now updated, so need to merge master and see if there is anything left to change

Done with rebasing and most of test cases are done only one remaining.

jreback commented:
thanks @krajatcl
